test/tests/val-2f/val-2f.i

```diff
@@ -0,0 +1,268 @@
+# Physical constants
+R = '${units 8.31446261815324 J/mol/K}' # ideal gas constant based on number used in include/utils/PhysicalConstants.h
+
+# Pressure conditions
+pressure_enclosure_init = '${units 13300 Pa}'
+pressure_enclosure_cooldown = '${units 1e-6 Pa}' # vaccum
+pressure_enclosure_desorption = '${units 1e-3 Pa}' # vaccum, assumed
+
+# Temperature conditions
+temperature_initial = '${units 773 K}'
+temperature_desorption_min = '${units 300 K}'
+temperature_desorption_max = '${units 1073 K}'
+temperature_cooldown_min = ${temperature_desorption_min}
+desorption_heating_rate = '${units ${fparse 3/60} K/s}'
+
+# Important times
+charge_time = '${units 50 h -> s}'
+cooldown_time_constant = '${units ${fparse 45*60} s}'
+# TMAP4 and TMAP7 used 40 minutes for the cooldown duration,
+# We use a 5 hour cooldown period to let the temperature decrease to around 300 K for the start of the desorption.
+# R.G. Macaulay-Newcombe et al. (1991) is not very clear on how long samples cooled down.
+cooldown_duration = '${units 5 h -> s}'
+desorption_duration = '${fparse (temperature_desorption_max-temperature_desorption_min)/desorption_heating_rate}'
+endtime = '${fparse charge_time + cooldown_duration + desorption_duration}'
+
+# Materials properties
+concentration_scaling = 1e10 # (-)
+diffusion_Be_preexponential = '${units 8.0e-9 m^2/s -> mum^2/s}'
+diffusion_Be_energy = '${units 4220 K}'
+solubility_order = .5 # order of the solubility law (Here, we use Sievert's law)
+solubility_constant_Be = '${fparse 7.156e27 / 1e18 / concentration_scaling}' # at/m^3/Pa^0.5 -> at/mum^3/Pa^0.5}
+solubility_energy_Be = '${units 11606 K}'
+solubility_constant_BeO = '${fparse 5.00e20 / 1e18 / concentration_scaling}' # at/m^3/Pa^0.5 -> at/mum^3/Pa^0.5}
+solubility_energy_BeO = '${units -9377.7 K}'
+
+# Numerical parameters
+dt_max_large = '${units 100 s}'
+dt_max_small = '${units 10 s}'
+dt_start_charging = '${units 1 s}'
+dt_start_cooldown = '${units 10 s}'
+dt_start_desorption = '${units 1 s}'
+
+# Geometry and mesh
+length_Be = '${units 0.4 mm -> mum}'
+length_Be_modeled = '${fparse length_Be/2}'
+num_nodes_Be = 40
+node_length_Be = '${fparse length_Be_modeled / num_nodes_Be}'
+
+[Mesh]
+  [cmg]
+    type = CartesianMeshGenerator
+    dim = 1
+
+    # Define cell lengths for only the Be subdomain
+    dx = '${node_length_Be} ${node_length_Be} ${node_length_Be} ${node_length_Be} ${node_length_Be}
+          ${node_length_Be} ${node_length_Be} ${node_length_Be} ${node_length_Be} ${node_length_Be}
+          ${node_length_Be} ${node_length_Be} ${node_length_Be} ${node_length_Be} ${node_length_Be}
+          ${node_length_Be} ${node_length_Be} ${node_length_Be} ${node_length_Be} ${node_length_Be}'
+
+    # Set all cells to subdomain 1 (now renumbered starting from 0)
+    subdomain_id = '1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1'
+  []
+[]
```

Comment:
This is a more complex mesh than what you need for this specific case, since you only have one material in this new validation case. Please simplify the mesh and make it relevant to the new case. 
Look at other cases to see how a simple 1D mesh can be generated much more simply.